### PR TITLE
- adding some documentation around the internal tls solution

### PR DESCRIPTION
**Pull request by gambol99**

### CODE REVIEW

File: how-to-docs/certificates.md [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+              fieldPath: metadata.namespace
+        volumeMounts:
+        # a emptyDir which the sidekicks writes the certificates
+        - name: certs
+          mountPath: /certs
+        # The platform CA Bundle hold the root ca used to verify the certificate chain
+        - name: bundle
+          mountPath: /etc/ssl/certs
+          readOnly: true
+      - name: you_application
```
Comment (vaijab) [inline how-to-docs/certificates.md]: `your_application`

File: how-to-docs/certificates.md [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+        image: quay.io/ukhomeofficedigital/some_image
+        ...
+        ports:
+        - name: https
+          port: 443
+          targetPort: 443
+        volumeMounts:
+        # You can configure you application to pick up the certificates from here (tls.pem and tls-key.pem)
```
Comment (vaijab) [inline how-to-docs/certificates.md]: your

File: how-to-docs/certificates.md [hunk 3]
```diff
@@ -0,0 +1,124 @@
+### **Application Certificates**
+-----
+
+The platform provide an in-house certificate authority for provide local hostname.namespace.svc.cluster.local certificates. Note: the CA is setup only to serve certificates under hostname.something.svc.clutser.local, ip addresses and single named hostnames, i.e. localhost, myservice etc.
```
Comment (jaykeshur) [inline how-to-docs/certificates.md]: provides an in-house certificate authority for provding

File: how-to-docs/certificates.md [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+
+#### **CA Bundle**
+
+By default in the all namespaces a ca bundle has been added which can been mounted into the /etc/ssl/certs of the container and contains the root CA used to verify authenticity of the certificates. An example of using it is given below.
```
Comment (jaykeshur) [inline how-to-docs/certificates.md]: By default, in all namespaces,

File: how-to-docs/certificates.md [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+      # Ensure you mount the volumes
+      volumes:
+      - name: bundle
+        configMap:
+          name: bundle
+      - name: certs
+        emptyDir: {}
+```
+
+The cfssl sidekick with request the certificates and place into the /certs directory which is shared between app and itself. Note, the ```--expiry`` can be used to configure the rotation of the certificatesl, though it's presentky up the application to reload them.
```
Comment (jaykeshur) [inline how-to-docs/certificates.md]: extra ` need after expiry. and presentky -> presently, and remove the extra l on certificates

File: how-to-docs/certificates.md [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+  type: ClusterIP
+  ports:
+  - name: https
+    port: 443
+    targetPort: 443
+  selector:
+    name: myservice
+```
+
+We then connect the services to together tieing up the ingress and backend services.
```
Comment (KashifSaadat) [inline how-to-docs/certificates.md]: "connect the services together, tying up.."

File: how-to-docs/certificates.md [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+
+```shell
+kind: Ingress
+metadata:
+  name: ingress
+  labels:
+    # this tell the ingress to use tls between itself and service
```
Comment (KashifSaadat) [inline how-to-docs/certificates.md]: "tells"

File: how-to-docs/certificates.md (hunk 5 above)
Comment (KashifSaadat) [inline how-to-docs/certificates.md]: "sidekick will request.."

File: how-to-docs/certificates.md [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+        - --domain=myservice.${KUBE_NAMESPACE}.svc.cluster.local
+        - --domain=another_domain_name
+        - --expiry=8760h
+        env:
+        - name: KUBE_NAMESPACE
+          valueFrom:
+            fieldRef:
+              fieldPath: metadata.namespace
+        volumeMounts:
+        # a emptyDir which the sidekicks writes the certificates
```
Comment (KashifSaadat) [inline how-to-docs/certificates.md]: "an emptyDir which the sidekick writes the certificates into"

**Comment by gambol99** [comment]
ok .. make the changes @jaykeshur @vaijab @KashifSaadat  :-) ...

**Pull request merged**